GoogleSignIn.podspec

```diff
@@ -33,9 +33,9 @@ The Google Sign-In SDK allows users to sign in with their Google account from th
   ]
   s.ios.framework = 'UIKit'
   s.osx.framework = 'AppKit'
-  s.dependency 'AppAuth', '~> 1.6'
-  s.dependency 'GTMAppAuth', '~> 4.0'
-  s.dependency 'GTMSessionFetcher/Core', '>= 1.1', '< 4.0'
+  s.dependency 'AppAuth', '~> 1.7.3'
```

Comment:
Just looked at the info about the optimistic operator. From my quick read, it seems like `'AppAuth', '~> 1.7'` would try to use the highest available version < 2.0, but could fall back to 1.7 as a minimum. For AppAuth, it looks like version 1.7.3 added a fix for SPM bundling -- would we want 1.7.3 as the minimum?

Reply:
If I'm reading the below from the docs correctly, then I think we want to stick with the `~> MAJOR.MINOR`. E.g., `s.dependency 'AppAuth', '~> 1.7'`. This will allow us to optimistically match updates up to but not including 2.0. That's what we want because we want to capture both minor and patch releases up to the next major version, which may break the API this version of GSI currently uses. 

If we do `s.dependency 'AppAuth', '~> 1.7.3'`, then I think we will not get changes in versions 1.8 - 2.0. LMK if I'm interpreting thing incorrectly.

```
* '~> 0.1.2' Version 0.1.2 and the versions up to 0.2, not including 0.2 and higher
* '~> 0.1' Version 0.1 and the versions up to 1.0, not including 1.0 and higher
* '~> 0' Version 0 and the versions up to 1.0, not including 1.0 and higher
```

Reply:
My concern is with the fact that `'AppAuth', '~> 1.7'` could potentially allow AppAuth 1.7.0, 1.7.1, 1.7.2 as valid versions to meet version restrictions if other pods have a max requirement on  AppAuth (if I understand correctly). But AppAuth 1.7.0, 1.7.1, and 1.7.2 have Privacy Manifest bundling issues. 

I do agree with your interpretation of `s.dependency 'AppAuth', '~> 1.7.3'`, i.e. it wouldn't get changes in versions 1.8 - 2.0. Maybe we need to move away from the optimistic operator in order and use `'>= 1.7.3', '< 2.0'`?

Reply:
Summarizing offline conversation: I'm fine with either so long as we have matching logic across SPM and CocoaPods. We talked about how scenarios where clients would get AppAuth 1.7.0 would ultimately be the result of pinning to that version for some reason or some other logic that would cause a downgrade to that version. IMHO that would be an error, and would either be caught at development time or upon submission (due to failing to find the privacy manifest).

Reply:
Resolving -- used the optimistic operator. Agree that devs can only get into a "bad" state by pinning to an (old) exact version of AppAuth, which should be avoided, and that errors will surface at some point in the pipeline

Reply:
`>= 1.7.3', '< 2.0'` may be a safer way to get what you want.